test/position-managers/Permit.t.sol

```diff
@@ -0,0 +1,313 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "forge-std/Test.sol";
+import {PoolManager} from "@uniswap/v4-core/src/PoolManager.sol";
+import {IPoolManager} from "@uniswap/v4-core/src/interfaces/IPoolManager.sol";
+import {IHooks} from "@uniswap/v4-core/src/interfaces/IHooks.sol";
+import {Currency, CurrencyLibrary} from "@uniswap/v4-core/src/types/Currency.sol";
+import {PoolId, PoolIdLibrary} from "@uniswap/v4-core/src/types/PoolId.sol";
+import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
+import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";
+import {StateLibrary} from "@uniswap/v4-core/src/libraries/StateLibrary.sol";
+
+import {IERC20} from "forge-std/interfaces/IERC20.sol";
+import {IERC721Permit} from "../../src/interfaces/IERC721Permit.sol";
+import {ERC721Permit} from "../../src/base/ERC721Permit.sol";
+import {UnorderedNonce} from "../../src/base/UnorderedNonce.sol";
+
+import {PositionConfig} from "../../src/libraries/PositionConfig.sol";
+import {IPositionManager} from "../../src/interfaces/IPositionManager.sol";
+
+import {PosmTestSetup} from "../shared/PosmTestSetup.sol";
+
+contract PermitTest is Test, PosmTestSetup {
```

Comment:
can we call this PermitERC721 ? Cause we will also have a permit for permit2 exposed

Reply:
will rename this at the end before merge to preserve reviews / diffs